packages/opentelemetry-core/src/default_span.ts

```diff
@@ -0,0 +1,106 @@
+/**
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  Span,
+  SpanContext,
+  Status,
+  Attributes,
+  SpanKind,
+} from '@opentelemetry/types';
+
+/** This is a no-op implementation of Span. */
+export class DefaultSpan implements Span {
+  protected name = 'DefaultSpan'; // default
+  protected spanId = '';
+  protected parentSpanId = '';
+  protected kind = SpanKind.INTERNAL; // default
+  protected spanContext: SpanContext | null = null;
+
+  constructor() {
+    // TODO: https://github.com/open-telemetry/opentelemetry-js/pull/35
+    // this.spanId = randomSpanId();
+  }
+
+  context(): SpanContext {
+    return this._context();
```

Comment:
do we need to prefix protected/private funs with underscore in TypeScript? I'm not that familiar with best practices just wondering as we have visibility keywords. (public/private/protected)

Reply:
(I see the same question later)

Reply:
(copied from above)

Underscore prefixes are require in this case, for a few reasons:

* This is a library and not an application. This means most users will be using JavaScript and not TypeScript. An underscore is the only way they have to know this should not be called directly.
* The underscore variants allow a simpler inheritance model, since some behavior can be included in the public versions (non-prefixed). I agree this could be solved with composition instead.